test/integration/framework/test_server.go

```diff
@@ -120,7 +129,11 @@ func StartTestServer(t testing.TB, setup TestServerSetup) (client.Interface, *re
 	if err != nil {
 		t.Fatalf("create temp file failed: %v", err)
 	}
-	defer os.RemoveAll(saSigningKeyFile.Name())
+
+	defer func() {
+		saSigningKeyFile.Close()
```

Comment:
is the `utiltesting.RemoveTestFile(t, f)` appropriate in here?  I don't see the files being removed.